docs/resources/aws_security_group.md.erb

```diff
@@ -12,27 +12,69 @@ SGs are a networking construct which contain ingress and egress rules for networ
+Resource parameters: group_id, group_name, id, vpc_id 
 
-    # Ensure you have a security group with a certain ID
+An `aws_security_group` resource block uses resource parameters to search for and then test a Security Group. If no SGs match, no error is raised, but the `exists` matcher returns `false`, and all scalar properties are`nil`. List properties returned under these conditions are empty lists. If more than one SG matches (due to vague search parameters), an error is raised.
+
+    # Ensure you have a security group with a specific ID
     # This is "safe" - SG IDs are unique within an account
     describe aws_security_group('sg-12345678') do
       it { should exist }
     end
 
-    # Ensure you have a security group with a certain ID
+    # Ensure you have a security group with a specific ID
     # This uses hash syntax
     describe aws_security_group(id: 'sg-12345678') do
       it { should exist }
     end
 
+    # Ensure you have a security group with a specific name.  Names are 
+    # unique within a VPC but not across VPCs.
+    # Using only group returns an error if multiple SGs match.
+    describe aws_security_group(group_name: 'my-group') do
+      it { should exist }
+    end
+    # Add vpc_id to ensure uniqueness. 
+    describe aws_security_group(group_name: 'my-group', vpc_id: 'vpc-12345678') do
+      it { should exist }
+    end
+
 <br>
 
 ## Examples
 
 The following examples show how to use this InSpec audit resource.
 
-As this is the initial release of `aws_security_group`, its limited functionality precludes examples.
+    # Ensure that the linux_servers group permits SSH from the 10.5.0.0/16 range, but not the world
+    describe aws_security_group(group_name: linux_servers) do
+      # This passes if any inbound rule exists that specifies 
+      # port 22 and the given IP range, regardless of protocol, etc.
+      it { should allow_in(port: 22, ipv4_range: '10.5.0.0/16') }
+
+      # This passes so long as no inbound rule that specifies port 22 exists
+      # with a source IP range of 0.0.0.0/0.  Other properties are ignored. 
+      it { should_not allow_in(port: 22, ipv4_range: '0.0.0.0/0') }
+
+      # This is an alternate way of expressing the previous test.
+      # be_open_to_the_world_on_port assumes you want inbound_rules.
+      it { should_not be_open_to_the_world_on_port(22) }
+    end
+
+    # Ensure that the careful_updates group may only initiate contact with specific IPs.
+    describe aws_security_group(group_name: 'careful_updates') do
+
+      # If you have two rules, with one CIDR each: 
+      [ '10.7.23.12/32', '10.8.23.12/32'].each do |allowed_destination|
+        # this doesn't care about which ports are enabled
+        it { should allow_out(ipv4_range: allowed_destination)}
+      end
+
+      # If you have one rule, with two CIDRs: 
+      it { should allow_out(ipv4_range: ['10.7.23.12/32', '10.8.23.12/32']}
```

Comment:
No spacing at all here and there is spacing above. What's the standard for the InSpec team?